api/v1alpha1/catalogentry_types.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	kcpv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/apis/conditions/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// InternalAPIBindingEntryLabelKey is the label key on an APIBinding with the
+	// base62(sha224(<workspace>:<exportName>)) as value to filter bindings by catalog entry.
+	InternalAPIBindingEntryLabelKey = "internal.catalog.kcp.dev/entry"
+)
+
+// These are valid conditions of CatalogEntry.
+const (
+	// CatalogEntryValid is a condition for CatalogEntry that reflects the validity
+	// of the referenced APIExport.
+	CatalogEntryValid conditionsv1alpha1.ConditionType = "CatalogEntryValid"
+	// CatalogEntryInvalidReferenceReason is a reason for the CatalogEntryValid
+	// condition of APIBinding that the referenced CatalogEntry reference is invalid.
+	CatalogEntryInvalidReferenceReason = "CatalogEntryInvalidReference"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// CatalogEntry is the Schema for the catalogentries API
+type CatalogEntry struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CatalogEntrySpec   `json:"spec,omitempty"`
+	Status CatalogEntryStatus `json:"status,omitempty"`
+}
+
+// CatalogEntrySpec defines the desired state of CatalogEntry
+type CatalogEntrySpec struct {
+	// references is a list of references to APIExports.
+	References []kcpv1alpha1.ExportReference `json:"references"`
+	// icon is the representation data that can be displayed in UI
+	// +optional
+	Icon Icon `json:"icon,omitempty"`
+	// description is a human-readable message to describe the information regarding
+	// the capabilities and features that the API provides
+	// +optional
+	Description string `json:"description,omitempty"`
+}
+
+// Icon is the graphical representation that can be displayed in UI component
+type Icon struct {
+	// data is the encoded icon data
+	Data []byte `json:"data"`
```

Comment:
@spadgett had a warning that directly embedding this will cause the `LIST` call for `CatalogEntry` to balloon in size.

Reply:
Embedding Icon base64 will make individual objects balloon in size, potentially to the point that the etcd size limit of an individual object comes into play. Is that cause for concern?

Reply:
I think that was exactly the concern, yes - can we get some clarification from prior experience here? @spadgett

Reply:
it's a concern. We could limit this to some max size, e.g. 64kb. Then we are save, and ppl don't use this as an image store. It should be about icons, icons of maybe 512x512 max in size. How big is a 512x512 image as png or jpg?

Reply:
Removed the `Icon` field for now until we can hash out a better way to manage Icon data.